components/handlers/module_view_in_ay_sem.py

```diff
@@ -71,12 +71,15 @@ def GET(self):
         '''
             Retrieve and render all the info of a module mounting
         '''
-        if SESSION['id'] != web.ACCOUNT_LOGIN_SUCCESSFUL:
+        if not session.validate_session():
             raise web.seeother('/login')
 
         input_data = web.input()
         module_code = input_data.code
         module_info = model.get_module(module_code)
+        if module_info is None:
```

Comment:
the 404 page is used in a few pages, would adding an error message be a good idea? is it going to be used in the final production?